### PR TITLE
Volume to namespace mapping on array

### PR DESCRIPTION
**Pull request by nidtara**
# Description
Namespace mapping to PV/PVC

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/272 |

# Checklist:

- [X] Have you run format,vet & lint checks against your submission?
- [x] Have you made sure that the code compiles?
- [X] Did you run the unit & integration tests successfully?
- [ ] Have you maintained at least 90% code coverage?
- [ ] Have you commented your code, particularly in hard-to-understand areas
- [ ] Have you done corresponding changes to the documentation
- [ ] Did you run tests in a real Kubernetes cluster?
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

### CODE REVIEW

File: service/step_defs_test.go [hunk 1]
```diff
@@ -2730,7 +2730,7 @@ func (f *feature) deletionWorkerProcessesWhichResultsIn(volumeName, errormsg str
 	if volumeID == "" {
 		return fmt.Errorf("Could not find volumeID for volume %s", volumeName)
 	}
-	volumeName = DeletionPrefix + "csi-" + f.service.opts.ClusterPrefix + "-" + volumeName
```
Comment (shanmydell) [inline service/step_defs_test.go:2733]: why this change is required?

Comment (nidtara) [reply service/step_defs_test.go:2733]: Since we are building the volume name here for the unit-tests, additional '-' to accommodate the namespace value

**Review by boyamurthy — COMMENTED** [review]
Please mock the namespace in unit-tests . Right now volume name does not contain namespace .
ex:  Created volume with ID: csi-TST-volume1--000197900046-976721017

**Review by shanmydell — DISMISSED** [review]

File: service/features/service.feature [hunk 2]
```diff
@@ -34,6 +34,11 @@ Feature: PowerMax CSI interface
       Given a PowerMax service
       And I call CreateVolume "volume1"
       Then a valid CreateVolumeResponse is returned
+ 
+    Scenario: Create volume with namespace
```
Comment (boyamurthy) [inline service/features/service.feature:38]: please add tag for the scenario so that it gets executed .

Comment (boyamurthy) [inline service/features/service.feature:38]: please add tag for the scenario so that it gets executed .

**Review by boyamurthy — COMMENTED** [review]
.

**Review by boyamurthy — APPROVED** [review]
LGTM.

**Review by shanmydell — APPROVED** [review]
